src/main/java/DNAnalyzer/CmdArgs.java

```diff
@@ -74,85 +76,36 @@ public class CmdArgs implements Runnable {
    */
   @Override
   public void run() {
-    try {
-      Main.clearTerminal();
-
-      final String dna = readDNA();
-      final List<String> proteins = new ProteinFinder().getProtein(dna, aminoAcid);
-      // Output the proteins, GC content, and nucleotide cnt found in the DNA
-      Properties.printProteinList(proteins, aminoAcid);
-      System.out.println("\nGC-content (genome): " + Properties.getGCContent(dna) + "\n");
-      Properties.printNucleotideCount(dna);
+    DNAnalyzer dnaAnalyzer = dnaAnalyzer(aminoAcid)
+      .isValidDna()
+      .replaceDNA("u", "t");
 
-      // Output the number of codons based on the reading frame the user wants to look
-      // at, and minimum and maximum filters
-      final CodonFrame codonFrame = new CodonFrame(dna, READING_FRAME, minCount, maxCount);
-      final ReadingFrames aap = new ReadingFrames(codonFrame);
-      System.out.println();
-      aap.printCodonCounts();
-
-      // Find protein sequence in DNA if necessary
-      readProtein().ifPresent(pr -> findProtein(dna, pr));
+    if (reverse) {
+      dnaAnalyzer = dnaAnalyzer.reverseDna();
+    }
 
-      // Find the longest protein in DNA
-      ProteinAnalysis.printLongestProtein(proteins);
+    dnaAnalyzer
+      .printProteins()
+      .outPutCodons(minCount, maxCount)
+      .printLongestProtein();
+  }
 
-      // Print if DNA is random
-      if (Properties.isRandomDNA(dna)) {
```

Comment:
@singloon it seems like you accidentally removed the call to isRandomDNA(). Could you please add this back?

Reply:
added it back

Reply:
cool

Reply:
Thanks for adding that back! It also seems that calling the nucleotide function returns an error - which is not present in main:
<img width="609" alt="image" src="https://user-images.githubusercontent.com/96280466/196824473-5d8ad642-bd90-4b33-a12c-0c43866b43f8.png">
Could you please look into that as well! Thank you so much for all of your help!!

Reply:
What's your input, the printNucleotideCount function hasn't changed.

When I test it seems fine
`java -jar build/libs/DNAnalyzer.jar assets/dna/random/dnalong.fa --amino=ser --min=10 --max=100
Proteins coded for Serine:
----------------------------------------------------
1. TCTCATGATCACCAGTTCTGACGTTACAGTATTTTCGGTTGAGCAGGCCCCATGGGGCCCCCGCATGCCGAATTACGATATGATGCCCACTATCCTGTGTCTTCCAACCTTATGACTGACTTGTATGCGCTGCGAGGTCCCTCGATAGATTTGCTCCCACCCGTCCCGGAAACCATATCGACGACTTGTAGGTCTCTAA
2. TCCTGTGTCTTCCAACCTTATGACTGACTTGTATGCGCTGCGAGGTCCCTCGATAGATTTGCTCCCACCCGTCCCGGAAACCATATCGACGACTTGTAGGTCTCTAA
3. TCAAAACCAATCTCATGATCACCAGTTCTGACGTTACAGTATTTTCGGTTGAGCAGGCCCCATGGGGCCCCCGCATGCCGAATTACGATATGATGCCCACTATCCTGTGTCTTCCAACCTTATGACTGACTTGTATGCGCTGCGAGGTCCCTCGATAGATTTGCTCCCACCCGTCCCGGAAACCATATCGACGACTTGTAGGTCTCTAA
4. TCGGTTGAGCAGGCCCCATGGGGCCCCCGCATGCCGAATTACGATATGATGCCCACTATCCTGTGTCTTCCAACCTTATGACTGACTTGTATGCGCTGCGAGGTCCCTCGATAGATTTGCTCCCACCCGTCCCGGAAACCATATCGACGACTTGTAGGTCTCTAA
5. AGTTCTGACGTTACAGTATTTTCGGTTGAGCAGGCCCCATGGGGCCCCCGCATGCCGAATTACGATATGATGCCCACTATCCTGTGTCTTCCAACCTTATGACTGACTTGTATGCGCTGCGAGGTCCCTCGATAGATTTGCTCCCACCCGTCCCGGAAACCATATCGACGACTTGTAGGTCTCTAA
6. AGCAGGCCCCATGGGGCCCCCGCATGCCGAATTACGATATGATGCCCACTATCCTGTGTCTTCCAACCTTATGACTGACTTGTATGCGCTGCGAGGTCCCTCGATAGATTTGCTCCCACCCGTCCCGGAAACCATATCGACGACTTGTAGGTCTCTAA

GC-content (genome): 0.16777216

Nucleotide count:
A: 25000381 (25.000381%)
T: 24998528 (24.998528%)
G: 25000967 (25.000969%)
C: 25000124 (25.000126%)

Codons in reading frame 1 (10-100 occurrences):
----------------------------------------------------

Longest gene (209 nucleotides): TCAAAACCAATCTCATGATCACCAGTTCTGACGTTACAGTATTTTCGGTTGAGCAGGCCCCATGGGGCCCCCGCATGCCGAATTACGATATGATGCCCACTATCCTGTGTCTTCCAACCTTATGACTGACTTGTATGCGCTGCGAGGTCCCTCGATAGATTTGCTCCCACCCGTCCCGGAAACCATATCGACGACTTGTAGGTCTCTAA

dnalong.fa has been detected to be random.`

Reply:
Huh, weird, it works now. I'm going to merge this PR.